java/src/main/native/src/ColumnVectorJni.cpp

```diff
@@ -176,7 +219,35 @@ JNIEXPORT jlong JNICALL Java_ai_rapids_cudf_ColumnVector_fromScalar(JNIEnv *env,
     cudf::mask_state mask_state =
         scalar_val->is_valid() ? cudf::mask_state::UNALLOCATED : cudf::mask_state::ALL_NULL;
     std::unique_ptr<cudf::column> col;
-    if (row_count == 0) {
+    if (dtype.id() == cudf::type_id::LIST) {
+      // Neither 'cudf::make_empty_column' nor 'cudf::make_column_from_scalar' supports
+      // LIST type for now (04/27/2021), so the list precedes the others and takes care
+      // of the empty column itself.
+      auto s_list = reinterpret_cast<cudf::list_scalar const *>(scalar_val);
+      cudf::column_view s_val = s_list->view();
+
+      // Offsets: [0, list_size, list_size*2, ..., list_szie*row_count]
+      auto zero = cudf::make_numeric_scalar(cudf::data_type(cudf::type_id::INT32));
+      auto step = cudf::make_numeric_scalar(cudf::data_type(cudf::type_id::INT32));
+      zero->set_valid(true);
+      step->set_valid(true);
+      static_cast<ScalarType *>(zero.get())->set_value(0);
+      static_cast<ScalarType *>(step.get())->set_value(s_val.size());
+      std::unique_ptr<cudf::column> offsets = cudf::sequence(row_count + 1, *zero, *step);
+
+      // Data:
+      // 'cudf::fill' does not support LIST type for now(04/27/2021), so builds the data
```

Comment:
Link to cudf issue rather than listing a date

Reply:
Removed the date.